### PR TITLE
[AzureMonitorDistro] fix for LiveMetrics Dependencies

### LINKED ISSUE #44889: [BUG] Start an internal activity raise a Dependency Call Failure in Live metrics
### Library name and version

Azure.Monitor.OpenTelemetry.AspNetCore 1.2.0

### Describe the bug

when I start an internal activity, a dependency failure was raised without specifying a status
image

### Expected behavior

Shouldn't show failure

### Actual behavior

show failure

### Reproduction Steps

```csharp
  public static readonly ActivitySource TaskActivitySource  = new("task");

var activity = TaskActivitySource.StartActivity($"BackgroundTask test",
    ActivityKind.Internal, parentContext: default);

activity?.SetStatus(ActivityStatusCode.Ok);
activity?.Stop();
```
addservice
```csharp
services.AddOpenTelemetry().WithTracing(t => t.AddSource("task")).UseAzureMonitor();
```

### Environment

.NET 8.0.4

### PR DESCRIPTION
**Pull request by TimothyMothra**
Fixes: https://github.com/Azure/azure-sdk-for-net/issues/44889

Our mapping for LiveMetrics dependencies was incorrect.

- For known dependencies, the mappings didn't match Exporter.
- For unknown or manual dependencies, several fields were left unset.
   A user reported that `IsSuccess` was left as default value `false` and were incorrectly shown on the "Failed Dependencies" graph.

## Changes
- Update all dependency types to match Exporter.
- Set additional properties for manual/unknown dependencies.

## Notes
Application Insights defines mapping the DependencyTelemetry to DependencyDocument.
https://github.com/microsoft/ApplicationInsights-dotnet/blob/f5377f2e8f1d3ae557c1064e8028c31b34876f9c/WEB/Src/PerformanceCollector/PerformanceCollector/QuickPulseTelemetryProcessor.cs#L222

```csharp
private static ITelemetryDocument ConvertDependencyToTelemetryDocument(DependencyTelemetry dependencyTelemetry)
{
    ITelemetryDocument telemetryDocument = new DependencyTelemetryDocument()
    {
        Name = TruncateValue(dependencyTelemetry.Name),
        Success = dependencyTelemetry.Success,
        Duration = dependencyTelemetry.Duration,
        ResultCode = dependencyTelemetry.ResultCode,
        CommandName = TruncateValue(dependencyTelemetry.Data),
    };
}
```

### CODE REVIEW

**Comment by TimothyMothra** [comment]
Changed this to Draft while I'm investigating other mappings

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/LiveMetrics/DataCollection/DocumentHelper.cs [hunk 1]
```diff
@@ -118,42 +123,53 @@ internal static RemoteDependency ConvertToDependencyDocument(Activity activity)
...
 
-                    // The following "EXTENSION" properties are used to calculate metrics. These are not serialized.
-                    remoteDependencyDocument.Extension_IsSuccess = activity.Status != ActivityStatusCode.Error;
+                    var dbName = AzMonList.GetTagValue(ref liveMetricsTagsProcessor.Tags,, SemanticConventions.AttributeDbName)?.ToString();
+                    if (dbName != null)
+                    {
+                        remoteDependencyDocument.Properties.Add(new KeyValuePairString(SemanticConventions.AttributeDbName, dbName));
+                    }
                     break;
                 case OperationType.Rpc:
```
Comment (rajkumar-rangaraj) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/LiveMetrics/DataCollection/DocumentHelper.cs:152]: Do we need to check for this case? In your previous PR RPC related check was removed from the exporter.

Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/LiveMetrics/DataCollection/DocumentHelper.cs:152]: Hi Raj, this PR is in Draft, you can ignore changes here for now.

Answering your question; No we don't need this. I think this section should be removed. 
I made these changes earlier in the week. Today I'm pushing my changes to save my progress.
I need to follow up with you offline about some different changes.

**Comment by TimothyMothra** [comment]
@rajkumar-rangaraj this is ready for review

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md [hunk 2]
```diff
@@ -15,6 +15,9 @@
   `APPLICATIONINSIGHTS_CONNECTION_STRING` is present in `IConfiguration`.
   ([#45292](https://github.com/Azure/azure-sdk-for-net/pull/45292))
 
+* Fixed a bug in LiveMetrics that counted all manually created Dependencies as failures.
```
Comment (rajkumar-rangaraj) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md:18]: Is this the only issue that was fixed?

Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md:18]: No, I need to add another entry about updating all Dependencies to match the Exporter's mappings.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md [hunk 3]
```diff
@@ -15,6 +15,12 @@
   `APPLICATIONINSIGHTS_CONNECTION_STRING` is present in `IConfiguration`.
   ([#45292](https://github.com/Azure/azure-sdk-for-net/pull/45292))
 
+* Updated field mappings for sample telemetry sent to LiveMetrics.
```
Comment (vishweshbankwar) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md:18]: Should this be in other section? if not then, we need more information here as its not clear what the bug fix was.

Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md:21]: You're right, the "Other" section is a better fit for this

**Review by vishweshbankwar — APPROVED** [review]

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/LiveMetrics/DataCollection/DocumentHelper.cs [hunk 4]
```diff
@@ -118,42 +123,42 @@ internal static RemoteDependency ConvertToDependencyDocument(Activity activity)
             switch (liveMetricsTagsProcessor.ActivityType)
             {
                 case OperationType.Http:
-                    remoteDependencyDocument.Name = activity.DisplayName;
-                    remoteDependencyDocument.CommandName = AzMonList.GetTagValue(ref liveMetricsTagsProcessor.Tags, SemanticConventions.AttributeUrlFull)?.ToString();
+
+                    var httpUrl = AzMonList.GetTagValue(ref liveMetricsTagsProcessor.Tags, SemanticConventions.AttributeUrlFull)?.ToString();
+                    remoteDependencyDocument.Name = liveMetricsTagsProcessor.Tags.GetNewSchemaHttpDependencyName(httpUrl) ?? activity.DisplayName;
```
Comment (rajkumar-rangaraj) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/LiveMetrics/DataCollection/DocumentHelper.cs:128]: What is the issue with previous functionality? Please note live metrics sends data every second, minimal processing of attributes are the best. I recommend something like below.

AzMonList.GetTagValue(ref liveMetricsTagsProcessor.Tags, SemanticConventions.AttributeUrlFull)?.ToString() ?? activity.DisplayName

Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/LiveMetrics/DataCollection/DocumentHelper.cs:128]: The previous implementation set `remoteDependencyDocument.Name = activity.DisplayName;` and all other Dependency types do this.

But the goal with this PR was to match the mappings being made in the Exporter.
https://github.com/Azure/azure-sdk-for-net/blob/125bfbb68c585771c6c60f32e0542d45bfc20cfe/sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Customizations/Models/RemoteDependencyData.cs#L71-L72

Comment (rajkumar-rangaraj) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/LiveMetrics/DataCollection/DocumentHelper.cs:128]: I recommend to keep this simple, considering its not stored anywhere.

**Review by rajkumar-rangaraj — APPROVED** [review]

**Pull request merged**